sklearnex/tests/_utils_spmd.py

```diff
@@ -111,6 +111,13 @@ def _generate_clustering_data(
     return X_train, X_test
 
 
+def _generate_weights(n_samples, dtype=np.float64, random_state=42):
+    # Generates weights
+    gen = np.random.default_rng(random_state)
+    weights = gen.uniform(low=-0.3, high=+0.7, size=(n_samples)).astype(dtype)
+    return weights
```

Comment:
Please reuse `_generate_statistic_data`, and avoid code duplication. Just update `_generate_statistic_data` primitive like this:


```python
def _generate_statistic_data(n_samples, n_features=None, dtype=np.float64, random_state=42):
    # Generates statistical data
    gen = np.random.default_rng(random_state)
    size = (n_samples, n_features) if n_features else (n_samples)
    data = gen.uniform(low=-0.3, high=+0.7, size=size).astype(dtype)
    return data
```